packages/framework/fluid-static/src/types.ts

```diff
@@ -38,24 +38,9 @@ export type LoadableObjectClassRecord = Record<string, SharedObjectKind>;
  * In this case placing ISharedObjectKind fixed one usage and didn't break anything, and generally seems more likely to work than the reverse, so this is the order being used.
  * This is likely (a bug in TypeScript)[https://github.com/microsoft/TypeScript/issues/45809].
  */
-export type LoadableObjectClass<T extends IFluidLoadable = IFluidLoadable> =
+export type LoadableObjectKind<T extends IFluidLoadable = IFluidLoadable> =
 	| ISharedObjectKind<T>
-	| DataObjectClass<T>;
-
-/**
- * A class that has a factory that can create a `DataObject` and a
- * constructor that will return the type of the `DataObject`.
- *
- * @typeParam T - The class of the `DataObject`.
- * @privateRemarks
- * Having both `factory` and constructor is redundant.
- * TODO: It appears the factory is what's used, so the constructor should be removed once factory provides strong typing.
- */
-export interface DataObjectClass<T extends IFluidLoadable> {
```

Comment:
Moved to `aqueduct/src/data-objects/types.ts`

Reply:
Also modified so it's possible to implement with something other than a class object.

Reply:
Moving this was causing downstream build failures that I haven't been able to diagnose yet. Moving the type back to this package (and restoring the original typing of `createDataObjectKind` in `aqueduct` seems to resolve the issue, so I'm going to stick with that for now.

Reply:
I'll continue investigating and put up a separate PR once I've figured out what's going on here.